packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -132,6 +125,24 @@ class _SpriteAnimationWidgetState extends State<_SpriteAnimationWidget>
     });
   }
 
+  void _setupController() {
+    if (_controller != null) {
```

Comment:
unecessary null check

Reply:
Necessary on `didUpdateWidget`.

Reply:
you dont because you do `_controller?.`

Reply:
I thought you meant the if in the referred line. Changing.

Reply:
the if is precisely what I meant; you don't need to do:
```
if (a != null) a?.foo();
```
since
```
a?.foo()
```
already means exactly that.